core/src/config.rs

```diff
@@ -85,51 +82,70 @@ impl Default for NetworkConfiguration {
 }
 
 impl Configuration {
-    /// This method will build `Configuration` from a json *pretty* formatted file (without `:` in
-    /// key names).
+    /// Construct [`Self`] from a path-like object.
     ///
-    /// # Panics
-    /// This method will panic if configuration file presented, but has incorrect scheme or format.
+    /// The method has automatic de-duplication for items that are
+    /// represented as [`HashSet`]s, and errors if a duplicate is
+    /// found for [`crate::sumeragi::config::TrustedPeers`], i.e. if
+    /// two peers have the same public key.
     ///
     /// # Errors
-    /// This method will return error if system will fail to find a file or read it's content.
-    pub fn from_path<P: AsRef<Path> + Debug>(path: P) -> Result<Configuration> {
-        let file = File::open(path).wrap_err("Failed to open the config file")?;
+    /// - File not found.
+    /// - File found, but peer configuration parsing failed.
+    /// - File re-opening failed.
+    /// - Length of the array in raw json representation is different
+    /// to the lenght of the array in
+    /// [`self.sumeragi.trusted_peers.peers`], most likely due to two
+    /// (or more) peers having the same public key.
+    pub fn from_path<P: AsRef<Path> + Debug + Clone>(path: P) -> Result<Configuration> {
```

Comment:
what about other ways to add peers to the network? I think those should be covered as well

Reply:
Register peer already has this check. I'll see if it's possible to implement in loading from `env`, but it will be complicated/fragile.

Reply:
Custom Deserialize implementation?

Reply:
I didn't mean they should be covered here. If this check is there, then nothing else needs to be done I think

Reply:
I think that a custom deserializer is a better approach in general. This also addresses the point you brought up about changing the underlying datastructure.